app/commands/evaluate_sources.py

```diff
@@ -0,0 +1,270 @@
+                    ):
+                        return "FGDC"
+                except AttributeError:
+                    # possibly there is no "metainfo"
+                    return "N/A"
+        # means it may be an ISO19115-X format
+        if len(xml_tree.tag.split("}")) > 1:
+            tag = xml_tree.tag.split("}")[1]
+            if tag == "MI_Metadata":
+                return "ISO19115-2"
+            elif tag == "MD_Metadata":
+                return "ISO19115-1"
+        return "N/A"
+
+
+def evaluate_sources() -> None:
+    """
+    Function used to gather and then evaluate
+    harvest resources to check that they're healthy.
+    """
+    fieldnames = [
+        "source_id",
+        "source_title",
+        "source_type",
+        "source_url",
+        "organization_id",
+        "organization_title",
+        "organization_state",
+        "status_code",
+        "metadata_types",
+        "last_modified",
+        "server_type",
+    ]
+    sources = get_sources()
+    timed_out_sources = []
+    count = 0
+
+    logger.info("Starting Harvest Source Evaluation!")
+    with open("harvest-sources.csv", mode="w", newline="") as file:
+        writer = csv.DictWriter(file, fieldnames=fieldnames)
+        writer.writeheader()
+
+        for source in sources:
+            response = None
+            response = handled_request(
+                url=source["source_url"],
+                headers=HEADERS,
+                timeout=10,
+            )
+            metadata_types = ["N/A"]
+            status_code = "Timeout Error"
+            timestamps = []
+            last_modified = "N/A"
+            server_type = "N/A"
+
+            if response:
+                status_code = response.status_code
+                server_type = get_server_type(response.headers.get("Server", ""))
+                if response.ok:
+                    if source["source_type"] in ["waf", "waf-collection"]:
+                        # because the metadata type is not exposed in
+                        # the waf we need to look at one of the xml files
+                        xml_file_urls = traverse_waf(response.url)
+                        metadata_types = []
+                        for xml_file_url in xml_file_urls[:10]:
+                            xml_response = handled_request(
+                                url=xml_file_url, headers=HEADERS, timeout=10
+                            )
+                            if xml_response and xml_response.ok:
+                                if xml_response.headers.get("Last-Modified"):
+                                    timestamps.append(
+                                        xml_response.headers.get("Last-Modified")
+                                    )
+                                metadata_type = determine_metadata_type(xml_response)
+                                # if None, it's a type we don't currently work with.
+                                if metadata_type is not None:
```

Comment:
Should you be returning `None` from `determine_metadata_type` instead of `N/A` to fix this check?